File: algorithm/폰켓몬_최철원.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        Map<Integer,Integer> map = new HashMap<>();
        for(int val: nums){
            if(map.containsKey(val))    map.put(val,map.get(val)+1);
            else    map.put(val,1);
        }
        return Math.min(nums.length/2,map.size());
    }
}

File: algorithm/폰켓몬_최철원.cpp
#include <vector>
#include <map>
#include <cmath>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    map<int,int> m;
    for(int val : nums){
        if(m.find(val)==m.end())    m[val]=1;
        else    m[val]++;
    }
    return min(nums.size()/2,m.size());
}
